supervisor/host/network.py

```diff
@@ -209,20 +211,54 @@ async def update(self, *, force_connectivity_check: bool = False):
 
         await self.check_connectivity(force=force_connectivity_check)
 
+    async def create_vlan(self, interface: Interface) -> None:
+        """Create a VLAN interface."""
+        if interface.vlan is None:
+            raise RuntimeError("VLAN information is missing")
+        # For VLAN interfaces, check if one already exists with same ID on same parent
+        try:
+            self.sys_dbus.network.get(interface.name)
+        except NetworkInterfaceNotFound:
+            _LOGGER.debug(
+                "VLAN interface %s does not exist, creating it", interface.name
+            )
+        else:
+            raise HostNetworkError(
+                f"VLAN {interface.vlan.id} already exists on interface {interface.vlan.interface}",
+                _LOGGER.error,
+            )
+
+        settings = get_connection_from_interface(interface, self.sys_dbus.network)
+
+        try:
+            await self.sys_dbus.network.settings.add_connection(settings)
+        except DBusError as err:
+            raise HostNetworkError(
+                f"Can't create new interface: {err}", _LOGGER.error
+            ) from err
+
+        await self.update(force_connectivity_check=True)
+
     async def apply_changes(
         self, interface: Interface, *, update_only: bool = False
     ) -> None:
         """Apply Interface changes to host."""
         inet: NetworkInterface | None = None
```

Comment:
```suggestion
```
I'm very confused why is not yelling about this since you dropped all the `None` checks in the if conditions below. But there's no reason to do this anymore. `self.sys_dbus.network.get` will always return a `NetworkInterface` object or raise, it will never return `None`. You can remove this declaration and the type can be correctly inferred below.